var/spack/repos/builtin/packages/nmodl/package.py

```diff
@@ -27,6 +28,20 @@ class Nmodl(CMakePackage):
     depends_on('py-sympy@1.2:')
     depends_on('py-pyyaml@3.13:')
 
+    def get_cmake_args(self):
+        spec = self.spec
+        options = [] 
+        # installation with pgi fails when debug symbols are added
+        if '%pgi' in spec:
+            options.append('-DCMAKE_BUILD_TYPE=Release')
+        else:
+            options.append('-DCMAKE_BUILD_TYPE=RelWithDebInfo')
+
```

Comment:
Note that Cmake build system in Spack has `build_type` variant by default. So we have to do this bit differently in case user has set that variant from CLI. See `legion` package for reference.